### ISSUE TITLE
Cloning disk over USB 3.2 Gen 2x2 only reaches speed of 2.6Gbit/s instead of 20Gbit/s

### ISSUE THREAD

**Issue by Maggi9295** [rescuezilla/rescuezilla#514]
I'd like to use Rescuezilla to backup my SSD to an external one. However, I can't get Rescuezilla to use the full USB 3.2 Gen 2x2 speed, it is unbearably slow.

I have used Clonezilla in the past, and was able to clone the full 4TB SSD in half an hour, so just below 20Gbit/s or 150GB/min.
But after switching to Rescuezilla I only get a speed of 2.6Gbit/s or 20GB/min. It took almost **three** hours to complete the cloning process using Rescuezilla.
It doesn't matter what kind of compression (or none) I use, it is always stuck at about 20GB/min.
I also tried turning off legacy USB support in BIOS, no change.
Reconnected the external SSD several times to make sure it wasn't a bad contact, also tried the second USB C connector on my motherboard that only supports 10Gbit/s, always just 20GB/min.
This happened both in Rescuezilla version 2.4.2 (the one I used a while back) and the current one, 2.5.1.

I spent a couple of hours searching, but wasn't able to even just remotely find any indication what the problem might be. What really confuses me is that it works perfectly in Clonezilla.

- Processor: Intel core i9 14900k (Not yet RMA'd, but throttled in BIOS)
- Memory: Corsair Vengeance RGB 64GB (2x32GB) DDR5 6600MT/s CL32 CMH64GX5M2B6600C32
- Motherboard: Biostar Z790 Valkyrie with BIOS version Z79AC920BST (Not yet updated to the newest version)
- Internal SSD: WD_BLACK SN850X 4TB M.2 PCIe NVMe WDS400T2X0E
- External SSD: 4TB Lexar NM790 M.2 2280 PCIe 4.0 x4 3D-NAND TLC (LNM790X004T-RNNNG)
- External SSD Case: SilverstoneMS12 20Gbps SuperSpeed+ USB 3.2 Type-C to NVMe M.2 SSD enclosure incl. cable

**Comment by oam7575** [comment]
Almost sounds like a driver / kernel issue not detecting the full USB 3.2 speeds.

You indicate you have tested Rescuezilla 2.4.2 and 2.5.1
The release page : https://github.com/rescuezilla/rescuezilla/releases/tag/2.5.1 shows several versions of v2.5.1 based on different Ubuntu releases.

Can you test and confirm that this issue occurs on the Noble Numbat release : https://github.com/rescuezilla/rescuezilla/releases/download/2.5.1/rescuezilla-2.5.1-64bit.noble.iso

If possible could you also confirm which version of Clonezilla you used that supported your USB 3.2 Gen 2x2 speeds.

This would be helpful to confirm if its a kernel driver issue or something else.

**Comment by shasheene** [comment]
Yep I agree with oam7575 that it's probably a kernel driver issue. The latest Rescuezilla (v2.5.1) has the "Noble" based on Ubuntu 24.04 Noble, which has support for hardware support for hardware from approximately April 2024 when that Ubuntu version was released.

As noted elsewhere (https://github.com/rescuezilla/rescuezilla/issues/512#issuecomment-2415633789)

> Yep, drivers in newer Linux kernel coming with upcoming versions of Rescuezilla based on more recent versions of Ubuntu.
>
> Ubuntu 24.04 comes (current Rescuezilla v2.5.1 "Noble" release) comes with (quick google) kernel version 6.8.0.
> 
> Ubuntu 24.10 (future Rescuezilla release based on Ubuntu 24.10 Oracular comes with (quick google) Linux 6.11 kernel.
> 
> Quick search also [reveals](https://www.neowin.net/news/ubuntu-2410-oracular-oriole-arrives-with-latest-kernel-and-gnome-47/): 
> 
> > Ubuntu 24.10 launches with the Linux 6.11 kernel. According to Canonical, _this marks a shift in its kernel selection policy so that the newest available upstream Linux kernel is chosen_ so that users get the latest features and hardware support at release. Doing this also benefits silicon vendors, who can now easily plan for future kernel versions for Ubuntu releases and hardware enablement updates to LTS releases.
> 
> (_italic emphasis_ mine)
> 
> I'll be releasing at least one new version of Rescuezilla closer to the end of the year.

**Comment by Maggi9295** [comment]
First of all, thanks for the replies and excuse my late answer.

> Almost sounds like a driver / kernel issue not detecting the full USB 3.2 speeds.
> You indicate you have tested Rescuezilla 2.4.2 and 2.5.1 The release page : https://github.com/rescuezilla/rescuezilla/releases/tag/2.5.1 shows several versions of v2.5.1 based on different Ubuntu releases.
> Can you test and confirm that this issue occurs on the Noble Numbat release : https://github.com/rescuezilla/rescuezilla/releases/download/2.5.1/rescuezilla-2.5.1-64bit.noble.iso

The file you linked with the Noble Numbat version is actually the version I used in the first place :D


> If possible could you also confirm which version of Clonezilla you used that supported your USB 3.2 Gen 2x2 speeds.
> This would be helpful to confirm if its a kernel driver issue or something else.

Judging by my downloads folder, it was version 3.1.1, at least the filename is `clonezilla-live-3.1.1-27-amd64.iso`

 
> **Edit:** Further, if possible from both Clonezilla and Rescuezilla, drop to a command line shell and run the command :
>     * `lsusb -t`

Sure thing:
```
root@ubuntu:~# lsusb -t
/:  Bus 001.Port 001: Dev 001, Class=root_hub, Driver=xhci_hcd/16p, 480M
    |__ Port 002: Dev 002, If 0, Class=Audio, Driver=snd-usb-audio, 12M
    |__ Port 002: Dev 002, If 1, Class=Audio, Driver=snd-usb-audio, 12M
    |__ Port 002: Dev 002, If 2, Class=Audio, Driver=snd-usb-audio, 12M
    |__ Port 002: Dev 002, If 3, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 010: Dev 003, If 0, Class=Hub, Driver=hub/4p, 480M
        |__ Port 001: Dev 005, If 0, Class=Human Interface Device, Driver=usbhid, 12M
        |__ Port 001: Dev 005, If 1, Class=Human Interface Device, Driver=usbhid, 12M
        |__ Port 001: Dev 005, If 2, Class=Human Interface Device, Driver=usbhid, 12M
        |__ Port 002: Dev 006, If 0, Class=Human Interface Device, Driver=usbhid, 12M
        |__ Port 002: Dev 006, If 1, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 014: Dev 004, If 0, Class=Wireless, Driver=btusb, 12M
    |__ Port 014: Dev 004, If 1, Class=Wireless, Driver=btusb, 12M
/:  Bus 002.Port 001: Dev 001, Class=root_hub, Driver=xhci_hcd/8p, 20000M/x2
    |__ Port 001: Dev 004, If 0, Class=Mass Storage, Driver=uas, 20000M/x2
    |__ Port 004: Dev 002, If 0, Class=Mass Storage, Driver=usb-storage, 5000M
    |__ Port 008: Dev 003, If 0, Class=Hub, Driver=hub/4p, 10000M
```
Looks good so far I'd say, it should be the 20000M/x2 connection listed.
